src/networking.c

```diff
@@ -3440,6 +3485,22 @@ sds getAllClientsInfoString(int type, int hide_user_data) {
     return o;
 }
 
+sds getAllFilteredClientsInfoString(clientFilter *client_filter, int hide_user_data) {
+    listNode *ln;
+    listIter li;
+    client *client;
+    sds o = sdsnewlen(SDS_NOINIT, 500);
+    sdsclear(o);
```

Comment:
Do we need to clear this at init?

Reply:
Yes, I belive we do.

`sdsnewlen` with `SDS_NOINIT` creates a string of length 500 with undefined content. `sdsclear` sets the nul terminator and the length.

```c
void sdsclear(sds s) {
    sdssetlen(s, 0);
    s[0] = '\0';
}
```

Reply:
I followed suit from here: https://github.com/valkey-io/valkey/blob/unstable/src/networking.c#L3432
I don't think its needed though. Will check if its noop, then will remove it.